lua_examples/net_send_queue.lua

```diff
@@ -0,0 +1,46 @@
+------------------------------------------------------------------------------
+-- Net send queueing helper
+--
+-- See also
+--   https://nodemcu.readthedocs.org/en/dev/en/modules/net/#netsocketsend
+--
+-- Based on Vladimir Dronnikov's
+--   MQTT queuing publish helper
+--   https://github.com/dvv/nodemcu-thingies/blob/master/mqtt-queue-helper.lua
+--   LICENCE: http://opensource.org/licenses/MIT
+--
+-- Send arbitrary payload:
+--   local socket = net.createConnection(net.TCP, 0)
+--   socket:connect(port, ip)
+--   local net_send = dofile("net_send_queue.lua")(socket)
+--   net_send(payload1)
+--   net_send(payload2)
+--   net_send(payload3)
+--
+------------------------------------------------------------------------------
+do
+  -- cache
+  local shift = table.remove
+  -- queue handler
+  local make_sender = function(socket)
+    local queue = { }
+    local is_sending = false
+    local function send()
+      if #queue > 0 then
+        local item = shift(queue, 1)
+        socket:send(item, send)
+      else
+        is_sending = false
+      end
+    end
+    return function(...)
+      queue[#queue + 1] = ...
+      if not is_sending then
+        is_sending = true
+        send()
+      end
+    end
+  end
+  -- expose
+  return make_sender
```

Comment:
This should be a module in this case rather than a single function as we also need to expose read access to `#queue`  and maybe handle the "close on all sent" case.  Also this send option overwrites any `socket:on("sent")` and v.v. as they are equivalents.  So we need to have a caveat on this.  

I also thing that this will lock in  a socket resource because of the send self reference.  Need to check this one.  

Sorry, it's late and I am tired.  I don't mean to be too critical, but doing this is complicated.  If we are going to do this wrapper then it should be complete and correct.